README.md

```diff
@@ -23,30 +23,82 @@ need not run as root.
 
 Here's a simple example of creating a provider:
 
-    var d = require('dtrace-provider');
+```javascript
+var d = require('dtrace-provider');
 
-    var dtp = d.createDTraceProvider("nodeapp");
-    var p1 = dtp.addProbe("probe1", "int", "int");
-    var p2 = dtp.addProbe("probe2", "char *");
-    dtp.enable();	   
+var dtp = d.createDTraceProvider("nodeapp");
+var p1 = dtp.addProbe("probe1", "int", "int");
+var p2 = dtp.addProbe("probe2", "char *");
+dtp.enable();
+```
 
 Probes may be fired via the provider object:
 
-    dtp.fire("probe1", function(p) {
-        return [1, 2];
-    });
-    dtp.fire("probe2", function(p) { 
-        return ["hello, dtrace via provider", "foo"];
-    });
+```javascript
+dtp.fire("probe1", function() {
+    return [1, 2];
+});
+dtp.fire("probe2", function() {
+    return ["hello, dtrace via provider", "foo"];
+});
+```
 
 or via the probe objects themselves:
 
-    p1.fire(function(p) {
-      return [1, 2, 3, 4, 5, 6];
-    });
-    p2.fire(function(p) {
-      return ["hello, dtrace via probe", "foo"];
+```javascript
+p1.fire(function() {
+  return [1, 2, 3, 4, 5, 6];
+});
+p2.fire(function() {
+  return ["hello, dtrace via probe", "foo"];
+});
+```
+
+Note that `.fire()` takes a callback that returns the arguments to be
+provided when the DTrace probe actually fires. This allows you to call
+`.fire()` unconditionally when you want to fire the probe, but the
+callback will be invoked only when the DTrace probe is actually
+enabled. This allows you to create probes whose arguments might be
+expensive to construct, and only do any work when the probe is
+actually enabled. (Such as converting a large object to a string
```

Comment:
Nits: The parenthetical here isn't quite a sentence -- maybe: "Examples might include ..." instead of "Such as").  The comma is also unnecessary.